lib/asm_client.rb

```diff
@@ -0,0 +1,29 @@
+class AsmClient
+  def post(url, payload = {})
+    request :post, url, payload
+  end
+
+  def request(method, url, body = {})
+    resp = connection.send(method) do |req|
+      req.url url
+      req.headers['Authorization'] = "Token token=#{ENV['ASM_API_TOKEN']}"
+      req.headers['Content-Type'] = 'application/json'
+      req.body = body.to_json
+    end
+
+    log = ['  ', method, url, body.inspect, "[#{resp.status}]"]
+    if !resp.success?
+      log << resp.body.inspect
+    end
+    Rails.logger.info log.join(' ')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.